Core/HLE/sceAtrac.cpp

```diff
@@ -2061,15 +2120,11 @@ void _AtracGenarateContext(Atrac *atrac, SceAtracId *context) {
 	context->info.loopNum = atrac->loopNum;
 	context->info.loopStart = atrac->loopStartSample > 0 ? atrac->loopStartSample : 0;
 	context->info.loopEnd = atrac->loopEndSample > 0 ? atrac->loopEndSample : 0;
-	if (context->info.endSample > 0) {
-		// do not change info.state if this was not called at first time
-		// In Sol Trigger, it would set info.state = 0x10 outside
-		// TODO: Should we just keep this in PSP ram then, or something?
-	} else if (!atrac->data_buf) {
-		context->info.state = ATRAC_STATUS_NO_DATA;
-	} else {
-		context->info.state = atrac->bufferState;
-	}
+
+	// Note that we read in the state when loading the atrac object, so it's safe
+	// to update it back here all the time.  Some games, like Sol Trigger, change it.
+	// TODO: Should we just keep this in PSP ram then, or something?
```

Comment:
Yeah, preferably I guess, but unless games poke the value it probably doesn't matter too much..